frappe/model/document.py

```diff
@@ -808,7 +809,9 @@ def check_if_latest(self):
 			self.check_docstatus_transition(0)
 			return
 
-		if cstr(previous.modified) != cstr(self._original_modified):
+		if (cstr(previous.modified) != cstr(self._original_modified)) and not getattr(
```

Comment:
This is a red flag, code related to transactional sanity shouldn't have any exceptions built into it.